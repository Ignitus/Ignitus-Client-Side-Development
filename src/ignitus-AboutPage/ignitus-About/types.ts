export type IntroductionType = {
  title: string;
};
