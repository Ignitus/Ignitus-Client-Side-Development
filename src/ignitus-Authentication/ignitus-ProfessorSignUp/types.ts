import {authData} from '../../ignitus-Shared/ignitus-DesignSystem/ignitus-Templates/ignitus-Authentication';

export interface Props {
  signUpRequest: Function;
  signUpData: authData;
  clearPreviousSignUp: Function;
}
