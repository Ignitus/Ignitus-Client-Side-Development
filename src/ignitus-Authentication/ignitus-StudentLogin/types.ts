import { AuthData } from '../../ignitus-Shared';

export interface LogInProps {
  logInRequestUsingEmail: Function;
  logInRequestUsingUsername: Function;
  logInData: AuthData;
  clearPreviousLogin: Function;
}
