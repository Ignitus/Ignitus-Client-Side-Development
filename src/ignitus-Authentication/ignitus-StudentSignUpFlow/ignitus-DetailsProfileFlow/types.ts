import { AppIcon } from '../../../ignitus-Shared';

export type HeadingProps = {
  title: string;
  icon: AppIcon;
  fontStyle?: string;
};

export type TitleProps = {
  fontStyle?: string;
};

export type EducationProps = {
  university: string;
  address: string;
  batch: string;
};

export type IconProps = {
  color: string;
  background: string;
  size: string;
};
