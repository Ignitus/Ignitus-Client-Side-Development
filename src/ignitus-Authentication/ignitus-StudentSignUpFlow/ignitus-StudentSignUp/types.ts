import {authData} from '../../../ignitus-Shared/ignitus-DesignSystem/ignitus-Templates/ignitus-Authentication/index';

export interface Props {
  signUpRequest: Function;
  studentSignUpData: authData;
  clearPreviousSignUp: Function;
}
