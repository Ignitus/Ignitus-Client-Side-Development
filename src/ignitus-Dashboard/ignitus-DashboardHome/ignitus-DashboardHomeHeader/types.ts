export type Props = {
  name: string;
  Skeleton?: boolean;
};

export type StyleProps = {
  size?: string;
};
