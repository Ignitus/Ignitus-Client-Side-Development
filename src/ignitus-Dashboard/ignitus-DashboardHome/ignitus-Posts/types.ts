export type PropTypes = {
  imageLink: string;
  title: string;
  position?: string;
  place?: string;
  company?: string;
  description?: string;
  type?: string;
  jobType?: string;
};
