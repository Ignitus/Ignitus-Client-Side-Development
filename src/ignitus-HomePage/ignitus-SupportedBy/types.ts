export type Props = {
  data: string[];
  type: string;
  paragraph: string;
};
