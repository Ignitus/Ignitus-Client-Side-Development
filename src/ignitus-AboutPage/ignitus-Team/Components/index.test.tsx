import React from 'react';
import { render, mount } from 'enzyme';
import { Team } from './index';
import { GitHubDataType } from '../types';

React.useLayoutEffect = React.useEffect;
/*eslint-disable*/ 
const octocatGithubData: GitHubDataType = {
  login: 'octocat',
  id: 1,
  node_id: 'MDQ6VXNlcjE=',
  avatar_url: 'https://github.com/images/error/octocat_happy.gif',
  gravatar_id: '',
  url: 'https://api.github.com/users/octocat',
  html_url: 'https://github.com/octocat',
  followers_url: 'https://api.github.com/users/octocat/followers',
  following_url: 'https://api.github.com/users/octocat/following{/other_user}',
  gists_url: 'https://api.github.com/users/octocat/gists{/gist_id}',
  starred_url: 'https://api.github.com/users/octocat/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/octocat/subscriptions',
  organizations_url: 'https://api.github.com/users/octocat/orgs',
  repos_url: 'https://api.github.com/users/octocat/repos',
  events_url: 'https://api.github.com/users/octocat/events{/privacy}',
  received_events_url: 'https://api.github.com/users/octocat/received_events',
  type: 'User',
  site_admin: false,
  contributions: 1000,
  user_id: 1234,
};
/* eslint-enable */

let wrapper;
const props: any = {
  getContributorsData() {
    return [];
  },
  contributorsData: {
    presets: [octocatGithubData],
    isFetching: false,
  },
};
beforeEach(() => {
  wrapper = mount(<Team {...props} />);
});

describe('<Team />', () => {
  it('should be rendered', () => {
    const tree = render(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('should render correct title', () => {
    expect(wrapper.find('h2').text()).toBe('Plus Icon1 Contributors');
  });

  it('shoud render contributors data correctly', () => {
    expect(wrapper.find('PureTeam').length).toBe(1);
    expect(
      wrapper
        .find('PureTeam')
        .at(0)
        .props('contributors') === octocatGithubData,
    );
  });
});
