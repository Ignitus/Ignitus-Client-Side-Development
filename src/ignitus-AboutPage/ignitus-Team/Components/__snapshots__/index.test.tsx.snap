// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Team /> should be rendered 1`] = `
<div
  class="css-1hs3667"
>
  <div
    class="css-1uzyx05"
  >
    <div
      class="css-1hs3667"
    >
      <h2
        class="css-lcriaz"
      >
        <svg
          aria-label="Plus Icon"
          class="css-1p4zvgu"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Plus Icon
          </title>
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
          />
        </svg>
        1 Contributors
      </h2>
      <p
        class="css-gvbdwp"
      >
        This project was made possible by these contributors. ðŸŽ‰ ðŸŽ¨
      </p>
    </div>
    <div
      class="css-1guir8e"
    >
      <div
        class="css-16kcklx"
      >
        <a
          class="css-861jbu"
          href="https://github.com/octocat"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="avatar octocat"
            class="css-a07s8t"
            src="https://github.com/images/error/octocat_happy.gif"
            width="100%"
          />
        </a>
      </div>
      <a
        class="css-jgl6n1"
        href="https://github.com/Ignitus/Ignitus-client/graphs/contributors"
      >
        View All 
        <svg
          aria-hidden="true"
          aria-label="KeyBoardArrowDown"
          class="css-1p4zvgu"
          data-icon="KeyBoardArrowDown"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            KeyBoardArrowDown
          </title>
          <path
            d="M8.12 9.29L12 13.17l3.88-3.88c.39-.39 1.02-.39 1.41 0 .39.39.39 1.02 0 1.41l-4.59 4.59c-.39.39-1.02.39-1.41 0L6.7 10.7c-.39-.39-.39-1.02 0-1.41.39-.38 1.03-.39 1.42 0z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
